packages/patternfly-4/react-core/src/components/Nav/NavGroup.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/Nav/styles.css';
+import a11yStyles from '@patternfly/patternfly-next/utilities/Accessibility/styles.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+import NavToggle from './NavToggle';
+import { AngleRightIcon } from '@patternfly/react-icons';
+
+const propTypes = {
+  id: PropTypes.string.isRequired,
+  title: PropTypes.string.isRequired,
+  ariaLabel: PropTypes.string.isRequired,
+  expandable: PropTypes.bool.isRequired,
+  children: PropTypes.node.isRequired,
+  innerClassName: PropTypes.string.isRequired,
+  groupId: PropTypes.number.isRequired,
+  activeGroup: PropTypes.number.isRequired,
+  activeItem: PropTypes.number.isRequired
+};
+
+const NavGroup = ({
+  id,
```

Comment:
It seems the purpose for the id is for the 'aria-labeledby' attribute.  If that is the case we probably don't want to force the consumer to provide one.  We can generate the id similar to how it is don in the Modal Component.

Reply:
NavGroup is an internal component that is not exposed to consumer. It is used by NavList and passed all the props. id is not required and default is generated as you've mentioned

Reply:
Yep.  I see it now.  Looks good!